gobject-introspection: libc not automatically linked?

mach-kernel opened the pull request:
I don't know if this affects only me, however this and `glib2` die with the below error. @onre pointed out that adding `-lc` should fix it (but should also not be necessary). Is this patch the wrong thing to do? I don't have any issues with any other RSE packages and the use of `-lc`.

The errors:

```
/usr/people/mach/rpmbuild/BUILD/gobject-introspection-1.62.0/mips-sgug-irix/../girepository/cmph/buffer_entry.c:83: undefined reference to `memcpy'
/usr/people/mach/rpmbuild/BUILD/gobject-introspection-1.62.0/mips-sgug-irix/../girepository/cmph/buffer_entry.c:48: undefined reference to `free'
```

My env:

```
MSGVERB=text:action
NOMSGSEVERITY=1
DISTCC_HOSTS=10.0.1.5:3632,lzo,cpp/24 10.0.1.15:3632,lzo,cpp/24
PWD=/usr/people/mach/workspace/sgug-rse
LOGNAME=mach
MANPATH=/usr/sgug/share/man:/usr/share/catman/a_man:/usr/share/catman/g_man:/usr/share/catman/p_man:/usr/share/catman/u_man
TZ=EST5EDT
HOME=/usr/people/mach
LD_LIBRARYN32_PATH=/usr/sgug/lib32:/usr/lib32:/lib32:/usr/lib:/lib
TERM=xterm
USER=mach
NOMSGLABEL=1
SHLVL=1
PS1=[sgugshell \u@\h \W]\$
LC_ALL=C
PATH=/usr/sgug/distccmasqbin:/usr/sgug/bin:/usr/bin/X11:/usr/bin:/bin:/usr/sbin:/usr/bsd
MAIL=/usr/mail/mach
SSH_TTY=/dev/ttyq0
OLDPWD=/usr/people/mach
_=/usr/sgug/bin/env
```

danielhams commented:
Suggest we start by looking into why you are having issues with glib2 -> that builds fine for a few of us, so maybe that can be a place to start.

Can you build glib2 latest wip version (don't forget the patches!) and stick the full un-altered build log in a gist (or equiv)?

e.g.

```
rpmbuild -ba glib2.spec --nocheck 1>/tmp/glib2buildlog.txt 2>&1
```

And we begin looking a bit deeper. Ty.

danielhams commented:
Build log from my host is here:

https://gist.githubusercontent.com/danielhams/e4403152a7a084ef92f1d5cc47b6437b/raw/ad5999d3a790c042c4cf57a9c939d2ae96e63790/glib2buildlog.txt

mach-kernel commented:
My log had unrecongized target warnings where yours did not. I must have installed meson via `pip` and forgotten about it.  Thought the common denominator was meson, so I removed it from site-packages and such and cleaned out the rpm, then rebuilt and installed, glib builds properly now. Sorry about this, should have checked in the first place.
